File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode curr=head;
        ListNode n=head.next;
        while(n!=null)
        {
            if(n.val!=0 && n!=curr)
            {
                curr.val+=n.val;
            }
            else
            {
                curr.next=n.next;
                curr=curr.next;
                n=n.next;
            }
            if(n==null)
                break;
            n=n.next;
        }
        return head;
    }
}

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeNodes(struct ListNode* head){
  struct ListNode* curr=head;
   struct ListNode* temp=head->next;
    if(head==NULL)
        return NULL;
    while(temp!=NULL)
    {
        if(temp->val!=0 && temp!=curr)
        {
            curr->val+=temp->val;
            
        }
        else
        {   curr->next=temp->next;
            curr=temp->next;
         temp=temp->next;
        }
        if(temp==NULL)
            break;
        temp=temp->next;
    }
    return head;
}